src/main/java/picard/sam/AbstractAlignmentMerger.java

```diff
@@ -205,7 +234,8 @@ public AbstractAlignmentMerger(final File unmappedBamFile, final File targetBamF
                                    final SAMFileHeader.SortOrder sortOrder,
                                    final PrimaryAlignmentSelectionStrategy primaryAlignmentSelectionStrategy,
                                    final boolean addMateCigar,
-                                   final boolean unmapContaminantReads) {
+                                   final boolean unmapContaminantReads,
+                                   final UNMAPPING_READ_STRATEGY unmappingReadsStrategy) {
```

Comment:
So ... since this breaks the API and callers will have to add the new parameter anyway, why not just do away with the `unmapContaminantReads` parameter?  Also, since `unmappingReadsStrategy` only applies to highly-clipped likely-contaminant reads, could that be reflected in the name.  E.g. `contaminantUnmappingStrategy`?